overlord/snapstate/snapstate.go

```diff
@@ -2145,6 +2260,64 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 	return updated, updateTss, nil
 }
 
+type updateGroup struct {
+	infos []minimalInstallInfo
+	names []string
+}
+
+func splitEssentialUpdates(deviceCtx DeviceContext, updates []minimalInstallInfo, names []string) (essential, nonEssential *updateGroup) {
+	var snapd minimalInstallInfo
+	var bases []minimalInstallInfo
+	essential = &updateGroup{}
+	nonEssential = &updateGroup{}
+
+	addInfo := func(grp *updateGroup, info minimalInstallInfo) {
+		grp.infos = append(grp.infos, info)
+		if len(names) != 0 {
+			name := info.InstanceName()
+			if strings.Contains(name, "_") {
+				name = name[:strings.IndexRune(name, '_')]
+			}
+			grp.names = append(grp.names, info.InstanceName())
+		}
+	}
+
+	for _, update := range updates {
+		switch update.Type() {
+		case snap.TypeSnapd:
+			snapd = update
+		case snap.TypeBase:
+			bases = append(bases, update)
+		case snap.TypeGadget, snap.TypeKernel:
+			// snaps that require a reboot
+			addInfo(essential, update)
+		default:
+			addInfo(nonEssential, update)
+		}
+	}
+
+	if snapd != nil {
+		// if there's no other essential snaps, group snapd with the apps/bases
+		// (order doesn't matter here, we sort later)
+		if len(essential.infos) > 0 {
+			addInfo(essential, snapd)
+		} else {
+			addInfo(nonEssential, snapd)
+		}
+	}
+
+	for _, base := range bases {
+		// the base is refreshed w/ essential snaps if there are any
+		if base.InstanceName() == deviceCtx.Base() && len(essential.infos) > 0 {
```

Comment:
In hybrid systems this is not really a boot base, so it would not trigger a reboot. Not sure then if it needs to be added to the essential list.

Reply:
Fixed this now